pkgs/tools/graphics/obj2voxel/default.nix

```diff
@@ -0,0 +1,52 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, static ? stdenv.hostPlatform.isStatic
+, config
+, cmake
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "obj2voxel";
+  version = "1.3.4";
+
+  src = (
+    fetchFromGitHub {
+      owner = "Eisenwave";
+      repo = finalAttrs.pname;
+      rev = "refs/tags/v${finalAttrs.version}";
+      fetchSubmodules = true;
+      hash = "sha256-FkGQjYm15le1Na+2U1TgK4nNOZNs5bME5J87pjVFpDs=";
+    }
+  ).overrideAttrs (
+    _: {
+      GIT_CONFIG_COUNT = 1;
+      GIT_CONFIG_KEY_0 = "url.https://github.com/.insteadOf";
+      GIT_CONFIG_VALUE_0 = "git@github.com:";
+    }
+  );
+
+  patches = [
+    # pull missing definitions (https://github.com/Eisenwave/voxel-io/pull/3)
+    ./stdlib-defs.patch
```

Comment:
Could you use `fetchpatch` with the url https://github.com/Eisenwave/voxel-io/pull/3.patch instead of commiting the file ?

Reply:
I have to look into that. Actually, I first thought about pulling the patch from the commit in the PR mentioned, but it's for a submodule, so it would have to be applied to a subdirectory of the source tree.

Reply:
Its not a blocker. But when patches are available in a more or less stable url, then putting them in the tree is not really needed.

Reply:
Please add the githubId. It helps if you ever change your GitHub handle.

Reply:
Definitely add github. Otherwise you can't be pinged by @ofborg when someone makes a change to this package.

Reply:
@nagy If I figure out a clean way to apply an external patch to a subdirectory of the source tree, I would actually prefer getting it from the PR. This would avoid duplicated effort since the build errors on aarch64-darwin I see through ofborg should also be pushed upstream.

Reply:
@dotlambda The more likely scenario would be that I delete my Github handle once I figure out how to deal with various vendor lock-in issues that prevent this ATM.

Reply:
`fetchpatch` can add a prefix

Reply:
I moved most patches to corresponding upstream PRs (https://github.com/Eisenwave/voxel-io/pull/4, https://github.com/Eisenwave/voxel-io/pull/3) and switched to `fetchpatch`. The remaining patch is specific to `1.3.4` as the upstream head already fixed it.